### DISCUSSION TITLE
Tariff Compare entities are not updating

### DISCUSSION THREAD

**Discussion by cbjoldham** [BottlecapDave/HomeAssistant-OctopusEnergy#1117] [Q&A] [answered]
Hi, just noticed that my Tarrifs Comparison entities don't seem to have updated since the 5th of December, all the other items seem fine.

I've tried reloading that integration entry but no change. On thing I do note is that if I try and use the `configure` option I get an error of `Config flow could not be loaded: 400: Bad Request`

I've tried to have a good look through the docs and previous issues and haven't found any clues so not sure if I'm missing something obvious or I've found an issue.

I'm currently on 13.2.1 and I think I updated from 13.1.3 around about the same time period, but that's memory not digging into logs.

In the diagnostics there is an access error reported per meter, but I think this is expected? I have an Octopus Mini so I believe my consumption data is coming from that not the meter?

```
"device": "Failed to retrieve - Errors in request (https://api.octopus.energy/v1/graphql/): [{'message': 'Unauthorized.', 'locations': [{'line': 2, 'column': 2}], 'path': ['smartMeterTelemetry'], 'extensions': {'errorType': 'AUTHORIZATION', 'errorCode': 'KT-CT-1111', 'errorDescription': 'The viewer is not authorized to execute the query/mutation. Check authentication or roles/permissions.'}}]"
```

This snippet from.the diagnostic shows the 2 "stuck " items and one updating as expected:

```
"octopus_energy_a_free_electricity_sessions_data_last_retrieved": {
        "last_updated": "2024-12-11T15:54:32.957173+00:00",
        "last_changed": "2024-12-11T15:54:32.957173+00:00"
      },
      "octopus_energy_electricity_24e8062043_2000024413602_octoplus_free_electricity_session_baseline": {
        "last_updated": null,
        "last_changed": null
      },
      "octopus_energy_electricity_24e8062043_2000024413602_previous_accumulative_cost_flexible_octopus": {
        "last_updated": "2024-12-05T15:30:18.342506+00:00",
        "last_changed": "2024-12-05T15:30:18.342506+00:00"
      },
      "octopus_energy_electricity_24e8062043_2000024413602_previous_consumption_rates_flexible_octopus": {
        "last_updated": "2024-12-06T13:30:21.705332+00:00",
        "last_changed": "2024-12-06T13:30:21.705332+00:00"
      }
```

I have also manually checked that I can do the public API call for the tariff and rate and that seems fine - I did wonder at one point if the rate I was comparing had ended but `tariffs_active_at` seems current.

**Comment by BottlecapDave** [comment, marked as answer]
Hello. There was an issue with diagnostics which caused certain errors until the integration was reloaded. This has been rectified as part of https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/releases/tag/v13.3.0, which should fix the the first error you reported. If you continue to not see the tariff comparison updated, please see if you have any other errors in your HA logs. Also could you tell me the tariff code you're trying to compare. I'm also assuming your standard previous accumulative consumption sensors are correctly updating?

**Reply by cbjoldham** [reply]
Thanks for the response, I've just updated to `13.3.0` so will let you know.

I'm comparing with `E-1R-INTELLI-VAR-22-10-14-H`

Yes everything else seems to be updating fine, the "current" version on the energy dashboard and I use the "previous" version on the tarrif compare history graph

**Reply by cbjoldham** [reply]
Confirmed this morning that the comparison sensor now looks like it's moving in response to our consumption and not flatlined at the last known value 👍
